pinecone/index_connection.go

```diff
@@ -49,11 +56,88 @@ func newIndexConnection(in newIndexParameters) (*IndexConnection, error) {
 	return &idx, nil
 }
 
+// Close closes the connection to a Pinecone index.
+//
+// Returns a pointer to an IndexConnection object and an error if the connection cannot be closed.
+//
+// Parameters:
+//   idx: The IndexConnection object.
+//
+// Example:
+//  ctx := context.Background()
+//
+//  clientParams := pinecone.NewClientParams{
+//	  ApiKey:    getEnvVars("PINECONE_API_KEY"),
+//	  SourceTag: "your_source_identifier", // optional
+//	}
+//
+//  pc, err := pinecone.NewClient(clientParams)
+//	if err != nil {
+//		log.Fatalf("Failed to create Client: %v", err)
+//  }
+//
+//  idxs, err := pc.ListIndexes(ctx)
+//  if err != nil {
+//	  fmt.Println("Error:", err)
+//  }
+//
+//  idx, err := pc.Index(idxs[0].Host)
+//  if err != nil {
+//    fmt.Println("Error:", err)
+//	  return
+//	}
+//
+//  err = idx.Close()
+//	if err != nil {
+//	  fmt.Println("Error:", err)
+//  }
 func (idx *IndexConnection) Close() error {
 	err := idx.grpcConn.Close()
 	return err
 }
 
+// UpsertVectors indexes vectors into a Pinecone index.
+//
+// Parameters:
+//  - ctx: A context.Context object controls the request's lifetime,
+//  allowing for the request to be canceled or to timeout according to the context's deadline.
+//  - in: The vectors to index.
+//
+// Returns the number of vectors upserted and an error if the request fails.
+//
+// Example:
+//  ctx := context.Background()
+//
+//  clientParams := pinecone.NewClientParams{
+//	  ApiKey:    getEnvVars("PINECONE_API_KEY"),
+//	  SourceTag: "your_source_identifier", // optional
+//	}
+//
+//  pc, err := pinecone.NewClient(clientParams)
+//	if err != nil {
+//    log.Fatalf("Failed to create Client: %v", err)
+//  }
+//
+//  idxs, err := pc.ListIndexes(ctx)
+//
+//  idx, err := pc.Index(idxs[0].Host)
+//  if err != nil {
+//	  fmt.Println("Error:", err)
+//  }
+//
+// 	vectors := []*pinecone.Vector{
+//    {
+//     Id:     "abc-1",
+//    Values: []float32{1.0, 2.0},
+//     },
+//  }
+//
+//  count, err := idx.UpsertVectors(ctx, vectors)
+//  if err != nil {
+//	  fmt.Println("Error:", err)
```

Comment:
nit: Similar to previous feedback - anywhere we're doing an "Error:" let's look at adding a little more context around what failed so grepping the code examples is a bit quicker.

Noticed this issue across the comments in this file.